radicale: update to 3.4.0.

jason1987d opened the pull request:
#### Testing the changes
- I tested the changes in this PR: **YES**

#### Local build testing
- I built this PR locally for my native architecture, x86_64-libc

jason1987d commented:
I apologize for the failed tests, will get a commit with a template for pika as well shortly.

ahesford commented on srcpkgs/python3-pika/template:
```diff
@@ -0,0 +1,17 @@
+revision=1
+build_style=python3-pep517
+hostmakedepends="python3-setuptools"
+depends="python3"
```
This package ships tests; can they be run by `./xbps-src check`?

jason1987d replied:
```
=> radicale-3.3.3_1: running do_check ...
============================= test session starts ==============================
platform linux -- Python 3.13.1, pytest-8.3.1, pluggy-1.5.0
rootdir: /builddir/radicale-3.3.3
configfile: pyproject.toml
plugins: flake8-1.1.1, cov-5.0.0
collected 202 items

radicale/tests/test_auth.py ...................................          [ 17%]
radicale/tests/test_base.py ............................................ [ 39%]
.......................................................                  [ 66%]
radicale/tests/test_config.py ...............                            [ 73%]
radicale/tests/test_expand.py ....                                       [ 75%]
radicale/tests/test_rights.py ...........                                [ 81%]
radicale/tests/test_server.py .......                                    [ 84%]
radicale/tests/test_storage.py ............................              [ 98%]
radicale/tests/test_web.py ...                                           [100%]

=============================== warnings summary ===============================
radicale/tests/test_base.py: 96 warnings
radicale/tests/test_expand.py: 8 warnings
radicale/tests/test_storage.py: 16 warnings
  /builddir/radicale-3.3.3/radicale/app/report.py:174: DeprecationWarning: Testing an element's truth value will always return True in future versions.  Use specific 'len(elem)' or 'elem is not None' test instead.
    props: Union[ET.Element, List] = root.find(xmlutils.make_clark("D:prop")) or []

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
====================== 202 passed, 120 warnings in 6.76s =======================
```

ahesford replied:
That's `radicale`. The new `python3-pika` should pull in pytest and run checks unless there is a compelling reason to skip them.

jason1987d replied:
Looks like I might need to edit something in do_check step. Something cannot find a tests module.

```
__________ ERROR collecting tests/unit/threaded_test_wrapper_test.py ___________
ImportError while importing test module '/builddir/python3-pika-1.3.2/tests/unit/threaded_test_wrapper_test.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/lib/python3.13/importlib/__init__.py:88: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/unit/threaded_test_wrapper_test.py:15: in <module>
    from tests.wrappers import threaded_test_wrapper
E   ModuleNotFoundError: No module named 'tests'
=========================== short test summary info ============================
ERROR tests/acceptance/blocking_adapter_test.py
ERROR tests/unit/io_services_test_stubs_test.py
ERROR tests/unit/threaded_test_wrapper_test.py
!!!!!!!!!!!!!!!!!!! Interrupted: 3 errors during collection !!!!!!!!!!!!!!!!!!!!
============================== 3 errors in 0.43s ===============================
=> ERROR: python3-pika-1.3.2_1: do_check: 'PATH="${testdir}/usr/bin:${PATH}" PYTHONPATH="${testdir}/${py3_sitelib}" PY_IGNORE_IMPORTMISMATCH=1 ${make_check_pre} pytest3 ${testjobs} ${make_check_args} ${make_check_target}' exited with 2
=> ERROR:   in do_check() at common/build-style/python3-pep517.sh:36
[jason@box void-packages]$ ls masterdir/builddir/python3-pika-1.3.2/
CHANGELOG.md     MANIFEST.in  dist       pika            test-requirements.txt
CONTRIBUTING.md  README.rst   docs       pika.egg-info   testdata
HISTORY.md       RELEASE.md   examples   pylintrc        tests
LICENSE          build        nose2.cfg  pyproject.toml  utils
[jason@box void-packages]$ ls masterdir/builddir/python3-pika-1.3.2/tests
acceptance  base  misc  stubs  unit  wrappers
[jason@box void-packages]$ ls masterdir/builddir/python3-pika-1.3.2/tests/base
__init__.py  async_test_base.py
```

I did add all the dependencies to checkdepends, except attempting to track down distfiles for this bizarre 'codecov' module...

ahesford commented on srcpkgs/python3-pika/template:
```diff
@@ -0,0 +1,17 @@
+maintainer="Jason Elswick <jason@jasondavid.us>"
+license="BSD-3-Clause"
+homepage="https://github.com/pika/pika"
+distfiles="https://github.com/pika/pika/archive/refs/tags/${version}.tar.gz"
```
Prefer using `${PYPI_SITE}` unless there is a compelling reason.

ahesford commented:
`python3-nose` is a removed package, and the version still in the repositories won't function with Python 3.13 anyway.

jason1987d commented:
> `python3-nose` is a removed package, and the version still in the repositories won't function with Python 3.13 anyway.

Best course of action then? Set make_check=no for pika for this reason?

jason1987d commented:
At this point, I have set make_check=no for pika due to the nose dependency issues. radicale (now 3.4.0) itself tests and also runs just fine, I updated it on my own system and have been using it. I changed the source on the distfiles for radicale to pypi_site as well.

ahesford commented on srcpkgs/python3-pika/template:
```diff
@@ -0,0 +1,18 @@
+homepage="https://github.com/pika/pika"
+distfiles="${PYPI_SITE}/p/pika/pika-${version}.tar.gz"
+checksum=b2a327ddddf8570b4965b3576ac77091b850262d34ce8c1d8cb4e4146aa4145f
+make_check=no # requires python3-nose, which is deprecated in void
```
The package isn't deprecated, it's been removed.
```suggestion
make_check=no # Tests require dead "nose" package
```
